osquery/logger/plugins/aws_util.cpp

```diff
@@ -179,10 +191,47 @@ OsqueryFlagsAWSCredentialsProvider::GetAWSCredentials() {
                                    FLAGS_aws_secret_access_key);
 }
 
-OsqueryAWSCredentialsProviderChain::OsqueryAWSCredentialsProviderChain()
+Aws::Auth::AWSCredentials
+OsquerySTSAWSCredentialsProvider::GetAWSCredentials() {
+  // Grab system time in seconds since epoch for token experation checks
+  auto now = std::chrono::system_clock::now();
+  auto epoch = now.time_since_epoch();
+  auto seconds = std::chrono::duration_cast<std::chrono::seconds>(epoch);
+  // Pull new STS creds if we dont have them cached from a previous run,
+  // otherwise return cached STS creds
+  if (token_expire_time <= seconds.count()) {
+    // Create and setup a STS client to pull our temp creds
+    VLOG(1) << "Generate new AWS STS credentials.";
+    initAwsSdk();
```

Comment:
Ya, with the sts addition, it will get called every time the sts creds expire.  Ive had this running on several machines for ~30 days and everything is still trucking.